Vagrantfile

```diff
@@ -76,40 +71,27 @@ Vagrant.configure(2) do |config|
       echo "LOCAL_DB exists!"
     fi
 
-    cat <<-EOF | su - postgres -c psql
-    -- Create the database user:
-    CREATE USER payments WITH PASSWORD 'payments';
-
-    -- Create dev db:
-    CREATE DATABASE dev WITH OWNER=payments
-      LC_COLLATE='en_US.utf8'
-      LC_CTYPE='en_US.utf8'
-      ENCODING='UTF8'
-      TEMPLATE=template0;
-
-    -- Create test db:
-    CREATE DATABASE test WITH OWNER=payments
-      LC_COLLATE='en_US.utf8'
-      LC_CTYPE='en_US.utf8'
-      ENCODING='UTF8'
-      TEMPLATE=template0;
-	EOF
   SHELL
 
 
   ######################################################################
   # Add PostgreSQL docker container
   ######################################################################
-  #config.vm.provision "shell", inline: <<-SHELL
-  #  # Prepare PostgreSQL data share
-  #  sudo mkdir -p /var/lib/postgresql/data
-  #  sudo chown vagrant:vagrant /var/lib/postgresql/data
-  #SHELL
-  # Add PostgreSQL docker container
-  #config.vm.provision "docker" do |d|
-  #  d.pull_images "postgres"
-  #  d.run "postgres",
-  #    args: "-d --name postgres -p 5432:5432 -v /var/lib/postgresql/data:/var/lib/postgresql/data"
-  #end
+  config.vm.provision "shell", inline: <<-SHELL
+    # Prepare PostgreSQL data share
+    sudo mkdir -p /var/lib/postgresql/data
+    sudo chown vagrant:vagrant /var/lib/postgresql/data
+  SHELL
+  # Add the dev PostgreSQL docker container
+  # Note: the "d" essentially refers to the "docker" CLI command
+  # hence, d.pull_images -> "docker pull <image_name>"
+  config.vm.provision "docker" do |d|
+    d.pull_images "postgres"
+    d.run "postgres",
+      args: "-d --name payments-database -p 5432:5432 -v /var/lib/postgresql/data -e POSTGRES_USER=payments -e POSTGRES_PASSWORD=payments -e POSTGRES_DB=dev"
+  end
+
+  # add docker_compose file
+  config.vm.provision :docker_compose
 
```

Comment:
to make better use of this docker_compose should we just change the docker provision to only build the thing but not run it?